org/pr/milestone.ts

```diff
@@ -30,5 +30,24 @@ export default async () => {
     // Warn if the PR doesn't have a milestone
     if (currentPR.data.milestone == null) {
         warn("PR is not assigned to a milestone.");
+        return;
+    }
+
+    // Warn if the milestone is closing in less than 4 days
+    const warningDays = 4;
+    const warningThreshold = warningDays * 1000 * 3600 * 24; // Convert days to milliseconds
+    if (currentPR.data.milestone.due_on != null) {
+        const today = new Date();
+        let dueDate : Date;
+        dueDate = new Date();
+        dueDate.setTime(Date.parse(currentPR.data.milestone.due_on));
```

Comment:
Open question since I don't know much TypeScript so maybe it works as expected, but just wanted to raise the question: how does this handle DST and timezones? As I understand it, dueDate is in an ISO string in the API but `today = new Date()` likely use local TZ. Maybe the parsing of ISO date from API already takes all that into account and all is good but figured it would be worth pondering and checking.

Reply:
Anyone in the @Automattic/type-review team have any suggestion regarding this?

Not directly specific to type annotation, but hopefully you TypeScript pros have some good suggestions for us 🙇‍♂️

Reply:
This doesn't seem specific to TypeScript, but to JavaScript and Dates.

If I understand, we're parsing an ISO 8601 date string which may or may not include an offset. If it's coming from the API, I'd bet it does and it's UTC, but that seems outside our concern. If it's ISO 8601, we should be able to parse it correctly with [`Date.parse()`](https://developer.mozilla.org/en-US/docs/Web/Java;Script/Reference/Global_Objects/Date/parse) into a number representing the "number of milliseconds since January 1, 1970, 00:00:00 UTC." We do something similar with the current date ([`Date.now()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now) would have been a good alternative for that).

Using [`Date.getTime()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTime):

> getTime() always uses UTC for time representation. For example, a client browser in one timezone, getTime() will be the same as a client browser in any other timezone.

So, the milliseconds are compared to see if the difference is less than 4 days, we're comparing UTC-based millisecond offsets (apples vs. apples).

Seems fine to me.

Reply:
Thanks for confirming! 🎉